straxen/plugins/acqmon_processing.py

```diff
@@ -56,163 +137,308 @@ def compute(self, raw_records_aqmon):
+            result["veto_type"][vetos_seen:vetos_seen + n_vetos] = veto_name + 'veto'
+
+            vetos_seen += n_vetos
+
+        # Straxen deadtime is special, it's a start and stop with no data
+        # but already an interval so easily used here
+        artificial_deadtime = aqmon_hits[(aqmon_hits['channel'] ==
+                                          AqmonChannels.ARTIFICIAL_DEADTIME)]
+        n_artificial = len(artificial_deadtime)
+
+        if n_artificial:
+            result[vetos_seen:n_artificial]['time'] = artificial_deadtime['time']
+            result[vetos_seen:n_artificial]['endtime'] = strax.endtime(artificial_deadtime)
+            result[vetos_seen:n_artificial]['veto_type'] = 'straxen_deadtime_veto'
+            vetos_seen += n_artificial
+
         result = result[:vetos_seen]
-        result['time'] = np.clip(result['time'], start, end)
-        result['endtime'] = np.clip(strax.endtime(result), 0, end)
+        result['veto_interval'] = result['endtime'] - result['time']
         sort = np.argsort(result['time'])
         result = result[sort]
         return result
 
-
-@numba.njit
-def channel_select_(rr, ch):
+    def handle_starts_and_stops_outside_of_run(
+            self,
+            veto_hits_start: np.ndarray,
+            veto_hits_stop: np.ndarray,
+            chunk_start: int,
+            chunk_end: int,
+            veto_name: str,
+    ) -> typing.Tuple[np.ndarray, np.ndarray]:
+        """
+        We might be missing one start or one stop at the end of the run,
+        set it to the chunk endtime if this is the case
+        """
+        # Just for traceback info that we declare this here
+        extra_start = []
+        extra_stop = []
+        missing_a_final_stop = (
+                len(veto_hits_start)
+                and len(veto_hits_stop)
+                and veto_hits_start[-1]['time'] > veto_hits_stop['time'][-1])
+        missing_a_final_stop = (
+                missing_a_final_stop
+                or (len(veto_hits_start) and not len(veto_hits_stop)
+                    )
+        )
+        if missing_a_final_stop:
+            # There is one *start* of the //end// of the run -> the
+            # **stop** is missing (because it's outside of the run),
+            # let's add one **stop** at the //end// of this chunk
+            extra_stop = self.fake_hit(chunk_end)
+            veto_hits_stop = np.concatenate([veto_hits_stop, extra_stop])
+        if len(veto_hits_stop) - len(veto_hits_start) == 1:
+            # There is one *stop* of the //beginning// of the run
+            # -> the **start** is missing (because it's from before
+            # starting the run), # let's add one **start** at the
+            # //beginning// of this chunk
+            extra_start = self.fake_hit(chunk_start)
+            veto_hits_start = np.concatenate([extra_start, veto_hits_start])
+
+        something_is_wrong = len(veto_hits_start) != len(veto_hits_stop)
```

Comment:
we don't see them anymore, thats the greatest fix of this PR https://xe1t-wiki.lngs.infn.it/doku.php?id=xenon:j_angevaare:busy_logic_straxen

Otherwise, we will crash here https://github.com/XENONnT/straxen/blob/a3da24e9f88c3a9e190efc5afe0fbadf109fbf3d/straxen/plugins/acqmon_processing.py#L274

Reply:
It's probably not necessary, as this didn't happen before, but we could throw an error if a start is missing for something else than the HEV. This would mean a digitizer went busy *before* run start.